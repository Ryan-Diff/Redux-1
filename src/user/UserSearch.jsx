import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchUser, setSearch } from '../actions/userActions';

const UserSearch = () => {
   const dispatch = useDispatch();
   const userName = useSelector(state => state.search);

   const handleSubmit = () => {
       dispatch(fetchUser(userName));
   };

   const handleChange = ({ target }) => {
       dispatch(setSearch(target.value));
   };

   return (
       <>
        <input onChange={handleChange} />
        <button onClick={handleSubmit}>Search users</button>
       </>
   );
};

export default UserSearch;