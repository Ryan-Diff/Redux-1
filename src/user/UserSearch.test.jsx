import React from 'react';
import { render, cleanup } from '@testing-library/react';
import UserSearch from './UserSearch';
import { Provider } from 'react-redux';
import store from '../store';

describe('UserSearch component', () => {
  afterEach(() => cleanup());
  it('renders UserSearch', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <UserSearch />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});