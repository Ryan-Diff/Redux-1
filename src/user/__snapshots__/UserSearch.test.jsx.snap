// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearch component renders UserSearch 1`] = `
<DocumentFragment>
  <input />
  <button>
    Search users
  </button>
</DocumentFragment>
`;
